source/Mlos.Python/mlos/Optimizers/RegressionModels/unit_tests/TestRegressionEnhancedRandomForestModel.py

```diff
@@ -251,12 +251,13 @@ def test_lasso_categorical_predictions(self):
         x_train_df, y_train_df = self.generate_points_nonhierarchical_categorical_quadratic(num_train_x)
         rerf.fit(x_train_df, y_train_df)
 
-        num_categorical_levels_expected = len(x_train_df['x0'].unique()) * len(x_train_df['i0'].unique())
+        num_categorical_levels_expected = len(rerf.one_hot_encoder_adapter.get_one_hot_encoded_column_names())
         num_continuous_dimensions = 2  # x1 and x2
-        final_num_features = num_categorical_levels_expected - 1 + num_continuous_dimensions
+        final_num_features = num_categorical_levels_expected + num_continuous_dimensions
         polynomial_degree = self.model_config.max_basis_function_degree
         num_terms_in_polynomial_per_categorical_level = self.n_choose_k(polynomial_degree + num_continuous_dimensions, num_continuous_dimensions)
-        num_terms_in_polynomial = num_terms_in_polynomial_per_categorical_level * num_categorical_levels_expected
+        num_terms_in_polynomial = num_terms_in_polynomial_per_categorical_level * (num_categorical_levels_expected + 1)\
+                                  - len(rerf.categorical_zero_cols_idx_to_delete_)

```

Comment:
Can you maybe explain this in a comment?
Is that for polynomial interactions between different levels (which will always be zero)?

Reply:
Sorry.  That var name is poorly chosen.  Chose a more fitting name and added some comments.